src/servers/openai-mcp-server.ts

```diff
@@ -131,15 +132,44 @@ export class OpenAIDeepResearchMCPServer extends BaseMCPServer {
         const { id } = fetchInputSchema.parse(request.params.arguments);
         // id is of the form "<datasource-id>:<question>"
         const [datasourceId, question = ""] = id.split(":");
-        const answer = await this.getThoughtSpotService().getAnswerForQuestion(question, datasourceId, false);
+        const thoughtSpotService = this.getThoughtSpotService();
+        const answer = await thoughtSpotService.getAnswerForQuestion(question, datasourceId, false);
+        const image = await thoughtSpotService.getAnswerImagePNG(answer.session_identifier, answer.generation_number);
+        const imageBuffer = await image.arrayBuffer();
         if (answer.error) {
             return this.createErrorResponse(answer.error.message, `Error getting answer ${answer.error.message}`);
         }
 
+        let tokenUrl = "";
+        
+        // Generate token and store in KV store
+        if (!answer.error && this.ctx.env?.OAUTH_KV) {
+            console.log("[DEBUG] Storing token in KV");
+            const token = crypto.randomUUID();
+            const tokenData = {
+                sessionId: answer.session_identifier,
+                generationNo: answer.generation_number,
+                instanceURL: this.ctx.props.instanceUrl,
+                accessToken: this.ctx.props.accessToken
+            };
+            await putInKV(token, tokenData, this.ctx.env);
+            tokenUrl = `${this.ctx.env?.HOST_NAME}/data/img?token=${token}`;
+        }
+        let content = answer.data;
+        if (tokenUrl) {
```

Comment:
will tokenUrl be not available in any case ? If not we can remove this conditionality

Reply:
done